src/parser/production.rs

```diff
@@ -0,0 +1,24 @@
+use super::LexemeType;
+use super::{Token, TokenType};
+use lrpar::{Lexeme, NonStreamingLexer, Span};
+
+// caller MUST pay attention to the index out of bounds issue
```

Comment:
```suggestion
/// caller MUST pay attention to the index out of bounds issue
```
Change this warning to document string that `rustc` can render. So that caller will see it when invoking this function.